template/noxfile.py.jinja

```diff
@@ -55,11 +56,12 @@ def run_user_tests(session):
 @nox.session(name="coverage")
 def run_coverage(session):
     """Run the coverage tests and generate an XML report."""
-    session.posargs.append("--cov=src/{{project_slug}}")
-    session.posargs.append("--cov-report=xml")
+    session.install("setuptools", silent=False)
+    session.install("coverage", silent=False)
+    session.install("-e", ".[dev]", silent=False)
+    session.run("pytest", "--cov=src/{{project_slug}}", "--cov-report=xml", "tests/user_tests")
     run_user_tests(session)
```

Comment:
I think this is not needed, as things stand. Things can probably be refactored to reuse code, my implementation was a quick and dirty fix to get things running.